### PR TITLE
Check if the episode is a key of the searchResults before sending it

### PR DESCRIPTION
**Pull request by p0psicles**
Check if the episode is a key if the searchResults before sending it to cache.
* Add the season pack's as a valid search result.

### CODE REVIEW

File: sickbeard/search.py [hunk 1]
```diff
@@ -579,11 +579,16 @@ def searchProviders(show, episodes, manualSearch=False, downCurQuality=False, ma
...
-                    # If we have at least a result from one provider, it's good enough to be marked as result
-                    manual_snatch_results.append(results)
+            # Let's create a list with episodes that we where looking for
+            searched_episode_list = [episode_obj.episode for episode_obj in episodes]
+            # Add the -1 to also match season pack results
+            searched_episode_list.append(-1)
+            for searched_episode in searched_episode_list:
+                if searched_episode in [x for x, _ in searchResults.items()]:
+                    results = curProvider.cache.update_cache_manual_search(searchResults[searched_episode])
+                    if results is not None:
```
Comment (fernandog) [inline sickbeard/search.py]: Just "if results:"

Comment (p0psicles) [inline sickbeard/search.py]: yeah i'll change that

File: sickbeard/search.py [hunk 2]
```diff
@@ -579,11 +579,16 @@ def searchProviders(show, episodes, manualSearch=False, downCurQuality=False, ma
...
-            for searched_episode in episodes:
-                results = curProvider.cache.update_cache_manual_search(searchResults[searched_episode.episode])
-                if results is not None:
-                    # If we have at least a result from one provider, it's good enough to be marked as result
-                    manual_snatch_results.append(results)
+            # Let's create a list with episodes that we where looking for
+            searched_episode_list = [episode_obj.episode for episode_obj in episodes]
+            # Add the -1 to also match season pack results
+            searched_episode_list.append(-1)
+            episodes_search_result = [x for x, _ in searchResults.items()]
```
Comment (p0psicles) [inline sickbeard/search.py]: Moved it up, because I don't need to rebuild that list within the loop

Comment (medariox) [inline sickbeard/search.py]: ```episodes_search_result = [x for x, _ in searchResults.items()]```
What are you trying to do here? Looks a bit strange to me.

Comment (p0psicles) [inline sickbeard/search.py]: I'm creating a list of the Keys for the search results. The Keys are the mapped episodes or -1

Comment (medariox) [inline sickbeard/search.py]: Ok, in this case you can just do:
```episodes_search_result = searchResults.keys()```
It will have the same effect.

**Comment by OmgImAlexis** [comment]
Approved!

File: sickbeard/search.py [hunk 3]
```diff
@@ -583,12 +583,12 @@ def searchProviders(show, episodes, manualSearch=False, downCurQuality=False, ma
             searched_episode_list = [episode_obj.episode for episode_obj in episodes]
             # Add the -1 to also match season pack results
             searched_episode_list.append(-1)
-            episodes_search_result = [x for x, _ in searchResults.items()]
+            episodes_search_result = list(searchResults)
```
Comment (labrys) [inline sickbeard/search.py]: This is unnecessary, inefficient and increases memory usage

File: sickbeard/search.py [hunk 4]
```diff
@@ -583,12 +583,12 @@ def searchProviders(show, episodes, manualSearch=False, downCurQuality=False, ma
...
             for searched_episode in searched_episode_list:
-                if searched_episode in episodes_search_result:
-                    if curProvider.cache.update_cache_manual_search(searchResults[searched_episode]):
-                        # If we have at least a result from one provider, it's good enough to be marked as result
-                        manual_snatch_results.append(True)
+                if searched_episode in episodes_search_result and \
```
Comment (labrys) [inline sickbeard/search.py]: `if searched_episode in searchResults ...`

Comment (p0psicles) [inline sickbeard/search.py]: Aarg was overthinking things. Your right.

**Pull request merged**